### DISCUSSION TITLE
1.0.0a8

### DISCUSSION THREAD

**Discussion by shiroinekotfs** [shiroinekotfs/jupyter-cpp-kernel#35] [Announcements]
**Full Changelog**: https://github.com/shiroinekotfs/jupyter-cpp-kernel/compare/1.0.0a7...1.0.0a8

# TL;DR

## What's new?

In this version, there are plenty of updates in security, performance, and features:

### Removed features & Solved bugs

* Removed `magics` features. That means you cannot use external magic to manipulate the `g++` compiling process
* Solved #29 
* Solved `getline()` capture error when using it with `ifstream`. That makes the file I/O much easier

### New features & Modifications

* `jdisplay` is now `_Jupyter_Display`. It creates less confusion and extends the compatibility of the kernel
* `algorithm` and `iomanip` are now in the default header list. No longer need to use `#include` with these libraries.
* More display methods in #33 are now supported. You can view these in [this documentation](https://github.com/shiroinekotfs/jupyter-cpp-kernel-doc)
* New logo

![image](https://github.com/shiroinekotfs/jupyter-cpp-kernel/assets/115929530/d8ff18f3-ac45-4134-a08b-9fbc2bed87ed)

### Windows

You have to install [GNU Compiler Collection for Windows](https://github.com/shiroinekotfs/jupyter-cpp-kernel/blob/master/INSTALL_ON_WINDOWS.md) before doing any additional installation

### Linux/macOS

You have to install GNU Compiler Collection that is compatible with your OS. You may need to go search for yourself.

### Windows

```cmd
python -m pip install --upgrade jupyter-cpp-kernel
```

### Linux/macOS

```shell
sudo pip install --upgrade jupyter-cpp-kernel
# Or
pip install --upgrade jupyter-cpp-kernel
```

## What's Changed
* Update to version 1.0.0a8 by @shiroinekotfs in https://github.com/shiroinekotfs/jupyter-cpp-kernel/pull/34

<hr /><em>This discussion was created from the release <a href='https://github.com/shiroinekotfs/jupyter-cpp-kernel/releases/tag/1.0.0a8'>1.0.0a8</a>.</em>

**Comment by anhvlttfs** [comment]
What I believe is that the cat logo takes lots of time to design and surely it's more than the kernel itself, correct?

![image](https://github.com/user-attachments/assets/ccaa88b8-4008-450e-ae9d-e249946d68e6)
